xdsl/dialects/csl/csl_stencil.py

```diff
@@ -121,6 +141,237 @@ def __init__(
         )
 
 
+@irdl_op_definition
+class ApplyOp(IRDLOperation):
```

Comment:
I'm not 100% sure I understand why this is a single op with 2 regions, rather than something like this:

```mlir
%reduced = csl_stencil.chunk_reduce(%args) {
	...
}
%res = csl_stencil.post_process(%reduced) {
	...
}
```


I'm not proposing to change this, I'm just confused..

Reply:
I was hoping it would be helpful to preserve the semantics of the `stencil.apply`, since it essentially splits its body into two parts. There is also some implied interaction between the two regions, where the result yielded by the one is an input arg to the other. Finally, the whole thing would likely get lowered to one API call (with two callbacks), so I was wondering if it'd just be easier. I honestly hadn't actually considered splitting up the op entirely.